apps/estuary/lib/estuary/message_handler.ex

```diff
@@ -1,8 +1,33 @@
 defmodule Estuary.MessageHandler do
+  @moduledoc """
+  This module will read an event from the event stream and persist it.
+  """
   use Elsa.Consumer.MessageHandler
   require Logger
 
   def handle_messages(messages) do
+    Enum.each(messages, fn message ->
+      with {:ok, body} <- message.value |> Jason.decode(),
+           %{"author" => author, "create_ts" => create_ts, "data" => data, "type" => type} <- body do
+        Estuary.EventTable.insert_event(
+          author,
+          create_ts,
+          data,
+          type
+        )
+      else
+        {:error, %Jason.DecodeError{}} ->
+          Yeet.process_dead_letter("", message, "estuary",
+            reason: "event's JSON could not be decoded"
+          )
+
+        bad_keys ->
+          Yeet.process_dead_letter("", message, "estuary",
+            reason: "event #{inspect(bad_keys)} was decoded but did not have the right keys."
```

Comment:
Is it possible we're yeeting this message because we couldn't decode the json?

Reply:
No it would match the above condition if that was the case.